pkg/virt-controller/watch/virt-metadata/virt-metadata.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package virt_metadata
+
+import (
+	"encoding/json"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/util"
+)
+
+const AnnotationKey = "virt-metadata"
+
+type virtMeta struct {
+	NetworkPciMap map[string]string `json:"network-pci-map,omitempty"`
+}
+
+func NewVirtMeta(networkPciMap map[string]string) virtMeta {
+	return virtMeta{
+		NetworkPciMap: networkPciMap,
+	}
+}
+
+func SetPodVirtMetadata(pod *v1.Pod, virtMetadata virtMeta) error {
+	jsonString, err := marshalToString(virtMetadata)
+	if err != nil {
+		return err
+	}
+
+	if len(pod.Annotations) == 0 {
+		pod.Annotations = make(map[string]string)
+	}
+
+	pod.Annotations[AnnotationKey] = jsonString
+	return nil
+}
+
+func GetPodVirtMetadata(pod *v1.Pod) (virtMeta, error) {
+	jsonString, ok := pod.Annotations[AnnotationKey]
+	if !ok {
+		return virtMeta{}, fmt.Errorf("%s key not found on pod annotations", AnnotationKey)
+	}
+
+	var virtMetadata virtMeta
+	err := json.Unmarshal([]byte(jsonString), &virtMetadata)
+	if err != nil {
+		return virtMeta{}, err
+	}
+
+	return virtMetadata, nil
+}
```

Comment:
If I understand correctly this package is going to be used at virt-controller reconcile loop code,
I am not sure if those functions will be needed (maybe there are equivalents) or if this is the right place for them because it's not clear how it's going to be used.

Are you going to introduce how the virt-metadata package going to be used?

Reply:
well not in this PR but I was thinking something in the lines of 
```
func (c *VMIController) sync {
...
if !isTempPod(pod) && isPodReady(pod) {
...
virtMeta := virt_metadata.GetPodVirtMetadata(pod)
virtMeta.NetworkPciMap = DO_THE_MAP_THING()
virt_metadata.SetPodVirtMetadata(pod, virtMeta)
...
}
...
}
```

Reply:
@ormergi Can I resolve this?